src/domains/Auth.ts

```diff
@@ -0,0 +1,73 @@
+import { getCookie } from '../utils';
+import { AUTH_BASE_URL } from '../constants';
+
+export async function signup(email: string, name: string, password: string) {
+  const response = await fetch(`${AUTH_BASE_URL}/register`, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      email,
+      name,
+      password,
+    }),
+  });
+  const jsonResponse = await response.json();
+
+  if (!jsonResponse.accessToken) {
+    throw new Error(jsonResponse);
+  }
+}
+
+export async function login(email: string, password: string) {
+  const response = await fetch(`${AUTH_BASE_URL}/login`, {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({
+      email,
+      password,
+    }),
+  });
+  const jsonResponse = await response.json();
+
+  if (!jsonResponse.accessToken) {
+    throw new Error(jsonResponse);
+  }
+
+  document.cookie = `user_id=${jsonResponse.user.id}`;
+  document.cookie = `access_token=${jsonResponse.accessToken}`;
+}
+
+export function logout() {
+  document.cookie = `user_id=`;
+  document.cookie = `access_token=`;
+}
+
+export function isLoggedIn() {
+  const userId: string = getCookie('user_id');
+  const accessToken: string = getCookie('access_token');
+
+  return userId && userId !== '' && accessToken && accessToken !== '';
+}
+
+export async function getUser(): Promise<object> {
+  const userId: string = getCookie('user_id');
+  const accessToken: string = getCookie('access_token');
+
+  if (!isLoggedIn()) return undefined;
```

Comment:
```suggestion
  if (!isLoggedIn()) return undefined;
  
  const userId: string = getCookie('user_id');
  const accessToken: string = getCookie('access_token');
```
isLoggedIn 에서도 getCookie를 해주고 있기 때문에 조건문을 거치기 전에 또 getCookie 해 줄 필요는 없어 보여서 순서를 이렇게 변경해 보는건 어떨까요~? 🙂

Reply:
헉 불필요하게 중복되는 코드가 있었군요! 완전히 동의합니다
좋은 의견 감사합니다 🙂 [13473a2](https://github.com/woowacourse/javascript-vendingmachine/pull/71/commits/13473a26fcbfe6cf77c853d9bba3b68e6d40bc95)